packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRoot.ts

```diff
@@ -77,26 +82,62 @@ export async function unavailableBeaconBlobsByRoot(
   }
 
   // resolve missing blobs
-  const blobIdentifiers: deneb.BlobIdentifier[] = [];
+  const blobIdentifiersWithCommitments: (deneb.BlobIdentifier & {kzgCommitment: deneb.KZGCommitment})[] = [];
   const slot = block.message.slot;
+  const fork = config.getForkName(slot);
   const blockRoot = config.getForkTypes(slot).BeaconBlock.hashTreeRoot(block.message);
 
   const blobKzgCommitmentsLen = (block.message.body as deneb.BeaconBlockBody).blobKzgCommitments.length;
   for (let index = 0; index < blobKzgCommitmentsLen; index++) {
-    if (blobsCache.has(index) === false) blobIdentifiers.push({blockRoot, index});
+    if (blobsCache.has(index) === false) {
+      blobIdentifiersWithCommitments.push({
+        blockRoot,
+        index,
+        kzgCommitment: (block.message.body as deneb.BeaconBlockBody).blobKzgCommitments[index],
+      });
+    }
   }
 
-  let allBlobSidecars: deneb.BlobSidecar[];
-  if (blobIdentifiers.length > 0) {
-    allBlobSidecars = await network.sendBlobSidecarsByRoot(peerId, blobIdentifiers);
+  const networkReqIdentifiers: deneb.BlobIdentifier[] = [];
+  if (executionEngine !== undefined) {
```

Comment:
`IBeaconChain` always has `executionEngine` so it should not be an optional param